snuba/datasets/discover.py

```diff
@@ -115,10 +106,31 @@ def match_query_to_table(
                     if event_type == "transaction":
                         return EVENTS
 
-    if len(event_types) == 1 and event_types.pop() == "transaction":
+    if len(event_types) == 1 and "transaction" in event_types:
```

Comment:
This is now going to make queries like
`type=transaction AND group_id=123`
fail with HTTP500 since the query would go to transactions while before they were served by the events table, providing (incidentally) an empty result. 

Before applying this behavior change we should either monitor it or agree with the discover team. 
Also, if we select a table that will definitively cause a query error this should raise a subclass of snubal.query.exceptions.InvalidQueryException so snuba will return HTTP400 instead of 500.

Reply:
The reason this doesn't work on the transaction table seems to be due to the specific way that group ID is transformed in the transaction translator. I've fixed this in the latest commit here so the request should succeed as before. Let me know if you agree with the approach. We are still using the transaction table just mapping group_id to 0 everywhere it appears in the query.